doc/sphinxext/custom_roles.py

```diff
@@ -3,7 +3,15 @@
 
 def rcparam_role(name, rawtext, text, lineno, inliner, options={}, content=[]):
     rendered = nodes.Text('rcParams["{}"]'.format(text))
-    return [nodes.literal(rawtext, rendered)], []
+
+    rel_source = inliner.document.attributes['source'].split('/doc/', 1)[1]
+    levels = rel_source.count('/')
+    refuri = ('../' * levels +
+              'tutorials/introductory/customizing.html#matplotlib-rcparams')
```

Comment:
use an absolute link? ("/tutorials/...")

Reply:
when trying this locally, sphinx links to `file:///tutorials/introductory/customizing.html#matplotlib-rcparams`

Reply:
Perhaps set `refid` instead of `refuri`? (just throwing some ideas around -- but somewhere docutils does have the ability to do this, so we just need to find the right way to trigger it...)

Reply:
Nope. `refid='matplotlib-rcparams'` just links to the local anchor `#matplotlib-rcparams`.

I don't have the time to scan through the docutils docs and try out various stuff. I've posted the [question to stackoverflow](https://stackoverflow.com/questions/50937109/custom-sphinx-role-for-references). If there is no hint from there, I'd leave it at that.

Reply:
(I don't think this needs to be resolved to merge this PR -- it's quite unimportant.  The only remaining point for me is about the styling.)